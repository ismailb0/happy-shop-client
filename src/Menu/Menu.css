.menu-item-list > div {
  border-top: solid;
}
