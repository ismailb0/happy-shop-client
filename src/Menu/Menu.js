import React, { Component, Fragment } from 'react';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Collapse from '@material-ui/core/Collapse';
import ExpandLess from '@material-ui/icons/ExpandLess';
import ExpandMore from '@material-ui/icons/ExpandMore';


const MENU = {
  'Makeup': {
    'Cheek': [],
    'Face': [
      'Contouring',
      'Primer',
      'Tools',
      'Foundation',
      'Face Powder',
      'Cc & Bb Cream',
      'Concealer',
    ],
    'Eyes': [],
    'Lips': [],
    'Nails': [],
  }
};

class Menu extends Component {
  state = {
    'isMakeupOpen': false,
    'isCheekOpen': false,
    'isFaceOpen': false,
    'isEyesOpen': false,
    'isLipsOpen': false,
    'isNailsOpen': false
  };

  openItem = (itemName) => {
    this.setState({
      [`is${itemName}Open`]: !this.state[`is${itemName}Open`]
    });
  }

  getSubSubMenu = (subSubMenu) => {
    return (
      <List component="div" disablePadding>
        {
          subSubMenu.map((item) => {
            return (
              <ListItem key={item} button>
                <ListItemText inset primary={item} />
              </ListItem>
          )})
        }
      </List>
    )
  }

  getSubMenu = (subMenu) => {
    return (
      <List component="div" disablePadding>
        {
          Object.keys(subMenu).map((item) => {
            return (
              <Fragment key={item}>
                <ListItem key={item} button>
                  <ListItemText inset primary={item} onClick={() => this.openItem(item)} />
                  {subMenu[item].length !== 0 && (this.state[`is${item}Open`] ? <ExpandLess /> : <ExpandMore />)}
                </ListItem>
                {subMenu[item].length !== 0 && (
                  <Collapse in={this.state[`is${item}Open`]} timeout="auto" unmountOnExit>
                    {this.getSubSubMenu(subMenu[item])}
                  </Collapse>
                )}
              </Fragment>
          )})
        }
      </List>
    )
  }

  getMenu = (menuObject) => {
    return (
      <List component="nav">
        {
          Object.keys(menuObject).map((item) => {
            return (
              <Fragment key={item}>
                <ListItem key={item} button>
                  <ListItemText inset primary={item} onClick={() => this.openItem(item)} />
                  {this.state[`is${item}Open`] ? <ExpandLess /> : <ExpandMore />}
                </ListItem>
                {Object.keys(menuObject[item]).length !== 0 && (
                  <Collapse
                    in={this.state[`is${item}Open`]}
                    timeout="auto"
                    unmountOnExit
                  >
                    {this.getSubMenu(menuObject[item])}
                  </Collapse>
                )}
              </Fragment>
            )
          })
        }
      </List>
    )
  }

  render() {
    return this.getMenu(MENU);
  }
}

export default Menu;
