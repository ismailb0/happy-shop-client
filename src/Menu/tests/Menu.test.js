import React from 'react';
import Menu from '../Menu';

const ReactTestRenderer = require('react-test-renderer');

describe('<Menu />', () => {

  it('should match snapshot', () => {

    const component = ReactTestRenderer.create(
      <Menu />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
