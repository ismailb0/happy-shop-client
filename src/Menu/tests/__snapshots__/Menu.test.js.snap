// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshot 1`] = `
<nav
  className="MuiList-root-1 MuiList-padding-2"
>
  <div
    className="MuiButtonBase-root-17 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 MuiListItem-button-14"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiListItemText-root-20 MuiListItemText-inset-21"
      onClick={[Function]}
    >
      <span
        className="MuiTypography-root-26 MuiTypography-subheading-33 MuiListItemText-primary-23"
      >
        Makeup
      </span>
    </div>
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-62"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
    <span
      className="MuiTouchRipple-root-75"
    />
  </div>
</nav>
`;
